velox/functions/lib/Re2Functions.cpp

```diff
@@ -378,7 +496,32 @@ class LikeConstantPattern final : public VectorFunction {
  public:
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
-            RE2::Quiet) {}
+            RE2::Quiet) {
+    pattern_ = pattern;
+    patternKind_ = PatternKind::kGenericPattern;
+    reducedPatternLength_ = 0;
+
+    if (!escapeChar) {
+      auto patternKind = determinePatternKind(pattern_);
+      patternKind_ = patternKind.first;
+      reducedPatternLength_ = patternKind.second;
+    }
+  }
+
+  bool match(const StringView input) const {
+    switch (patternKind_) {
```

Comment:
This switch happens in a hot loop. I wonder if compiler is smart enough to recognize that patternKind_ is constant and pull it out of the loop. If not, we may need to do that manually. Would you compare the performance of this code vs. hard-coded match logic without a switch and a call to 'matchXxx' function?

Reply:
Thanks for the suggestion, modified LikeConstantPattern into a template parameterized on PatternKind as shared. There is an issue with the inputs for the benchmark (generated by Fuzzer/TpchGen), will share the performance comparison once this is fixed.

Reply:
There is not much performance difference seen by templating LikeConstantPattern on PatternKind:
```
| No. of runs | Pattern Kind | W/o templating LikeConstantPattern (s) | With templating LikeConstantPattern (s) | % Change |   |
|-------------|--------------|----------------------------------------|-----------------------------------------|----------|
|         500 | Fixed        |                                   3.78 |                                    3.65 |     3.44 |   
|             | Prefix       |                                   7.47 |                                    7.78 |    -4.15 |   
|             | Suffix       |                                   9.13 |                                    8.93 |     2.19 |   
|        1000 | Fixed        |                                   7.59 |                                     7.4 |      2.5 |   
|             | Prefix       |                                  14.96 |                                   15.08 |     -0.8 |   
|             | Suffix       |                                  18.01 |                                    18.1 |     -0.5 |   
|        1500 | Fixed        |                                  10.72 |                                   11.53 |    -7.56 |
|             | Prefix       |                                  22.16 |                                   23.05 |    -4.02 |   
|             | Suffix       |                                  26.58 |                                   29.12 |    -9.56 |
```
I figure the compiler recognizes patternKind_ is constant. Should I stick to the current code or revert back to the switch case?